subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyConstraintHandler.java

```diff
@@ -168,7 +168,7 @@ public DependencyConstraint create(Object dependencyNotation, Action<? super Dep
 
     @Override
     public DependencyConstraint enforcedPlatform(Object notation) {
-        DependencyConstraintInternal platformDependency = (DependencyConstraintInternal) create(notation);
+        DependencyConstraintInternal platformDependency = (DependencyConstraintInternal) doCreate(notation, null);
```

Comment:
This strikes me as odd, there isn't much saved in doing this, and it adds a second place that the default `null` is written out.

Reply:
I was wondering about it when changing it. But it seemed to make sense to go directly for internal APIs from public ones. But it is true that this instance does not resolve anything.

Reply:
I ended up reverting that change